lib/ansible/modules/commands/command.py

```diff
@@ -237,10 +247,23 @@ def main():
                 changed=False,
                 rc=0
             )
+        # if in check mode and the filename exists, report that a change would
+        # be made
+        elif module.check_mode:
+            module.exit_json(
+                cmd=args,
+                stdout="changed, since %s exists" % removes,
+                changed=True,
+                rc=0
+            )
 
     if warn:
         check_command(module, args)
 
+    # skip if in check mode so no changes are made
+    if module.check_mode:
+        module.exit_json(msg="skipped, running in check mode", skipped=True)
+
     startd = datetime.datetime.now()
 
     rc, out, err = module.run_command(args, executable=executable, use_unsafe_shell=shell, encoding=None, data=stdin)
```

Comment:
Instead of the checks inside of the removes and creates conditions above, conditionalize running this command on check_mode like this:
``` python
startd = datetime.datetime.now()

if not module.check_mode:
    rc, out, err = module.run_command(args, executable=executable, use_unsafe_shell=shell, encoding=None, data=stdin)
elif removes or creates:
    rc = 0
    stdout = stderr = b'Command would have run if not in check mode'
else:
    module.exit_json(msg="skipped, running in check mode", skipped=True)
```

Reply:
Thank you, that is clearer but I don't think this would be what we want.

This would not let the user know if it _would_ change or not, ie. it doesn't really add anything to what it currently does.

Reply:
I'll carry on the conversation here, closer to the code it references ...

I'm assuming that your suggestion means that lines 218-258 (in my current code) would be removed, otherwise the middle `elif removes or creates:` will never be true because its handled above. In that case, my current code differs in that it will actually check whether the file exists and will report a `changed` value accordingly, whereas your suggestion will return a blanket _maybe_ without actually checking.

I feel we might be getting wires crossed here so I'll reiterate my intentions just in case - I would like for this module to be able to handle check mode and use the provided creates/removes to return something other than the `skipped` it currently does.

Let me know if you were actually meaning for lines 218-258 to remain. If so, this still wouldn't have reduced the number of conditionals. Also let me know if I'm not clear enough about what I'd like the module to do. Sorry for any confusion :smile:

Reply:
Hmm I realised you were asking for an example (plus, I messed up the explanation anyway):

I use the module setting creates to some file it is supposed to create, I run the playbook in check mode.  
In your suggestion it reports as changed but it actually might not have been.  
In my current implementation it will report as changed if it doesn't exist and not if it does exist.

Reply:
Ahh yes I see now and it all makes sense!! I'll update the code now but the docs update for the shell module I'll have to do another time (hopefully tomorrow). Sorry to keep buggin you